bin2c.c

```diff
@@ -79,32 +89,52 @@ main(int argc, char *argv[])
     buf = bz2_buf;
 #endif
 
-    f_output = fopen(argv[2], "w");
+    filename = argv[arg++];
+    f_output = fopen(filename, "w");
     if (f_output == NULL) {
-        fprintf(stderr, "%s: can't open %s for writing\n", argv[0], argv[1]);
+        fprintf(stderr, "%s: can't open %s for writing\n", argv[0], filename);
         return -1;
     }
 
-    ident = argv[3];
+	// Check if array_name passed on command line
+	if (arg < argc) {
+		strcpy(array_name, argv[argc]);
```

Comment:
Should probably use `strncpy` here

Reply:
Alternatively, could `malloc` the correct size (`strdup` comes to mind). Just don't forget to free it at the end :D